cdi-unit/src/main/java/org/jglue/cdiunit/internal/easymock/EasyMockDiscoveryExtension.java

```diff
@@ -0,0 +1,34 @@
+package org.jglue.cdiunit.internal.easymock;
+
+import org.jglue.cdiunit.internal.DiscoveryExtension;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+
+public class EasyMockDiscoveryExtension implements DiscoveryExtension {
+
+	private Class<? extends Annotation> fieldAnnotation = null;
```

Comment:
@ikysil Is this not initialised immediately (and `final`) as a performance optimisation? If so, it would be good to have a comment to that effect (and ditto in `MockitoDiscoveryExtension`).

Reply:
We can not initialize the field eagerly because the lookup for the extension fails with NoClassDefFound error when EasyMock is not available. The non-null value here means that we have EasyMock in the classpath.

I will add a comment above to the code.

Also, this class is the annotation we want to process later.

Reply:
See commit [802f30c](https://github.com/cdi-unit/cdi-unit/commit/802f30c) for the corresponding changes.